clc/modules/object-storage/src/main/java/com/eucalyptus/objectstorage/ObjectStorageGateway.java

```diff
@@ -1158,7 +1168,13 @@ public SetObjectAccessControlPolicyResponseType setObjectAccessControlPolicy(fin
    */
   @Override
   public GetObjectResponseType getObject(final GetObjectType request) throws S3Exception {
-    ObjectEntity objectEntity = getObjectEntityAndCheckPermissions(request, request.getVersionId());
+    ObjectEntity objectEntity = null;
+    if (Strings.isNullOrEmpty(request.getVersionId()) || request.getVersionId().equalsIgnoreCase(ObjectStorageProperties.NULL_VERSION_ID)) {
```

Comment:
Hey @neilsoman! Just testing out a commit for a code review process, not going to merge this stuff. And yeah, the code change sucks, will implement it differently.